WordPress/Jetpack/Classes/NUX/New Landing Screen/Views/InfiniteScrollerView.swift

```diff
@@ -0,0 +1,114 @@
+import Foundation
+import UIKit
+
+/// A scroll view that produces an infinite scrolling effect.
+///
+/// Set the `scrollerDelegate` property to externally alter the scrolling rate speed and direction.
+///
+/// By default user interaction with the scroll view is disabled, but it can be enabled if desired.
+///
+/// Currently the view only supports vertical scrolling, but could easily be updated to support horizontal.
+/// This class also currently makes the assumption that the view generated by the `viewBuilder` closure will be taller
+/// than its frame.
+class InfiniteScrollerView: UIScrollView {
+    /// Delegate to poll for the speed and direction of the scrolling offset. Polled at the default rate of `CADisplayLink`.
+    var scrollerDelegate: InfiniteScrollerViewDelegate?
+
+    private var displayLink: CADisplayLink?
+
+    /// Closure used to repeat a view to populate a `UIStackView` for scrolling.
+    /// Provides the index of the view in the `UIStackView`.
+    private var viewBuilder: ((Int) -> UIView)?
+    private var stackView: UIStackView?
+
+    /// Returns the height of the first view in the stack view, if one exists. Used to compute reset points for vertical scrolling.
+    private var singleViewHeight: CGFloat? {
+        guard let stackView = stackView, !stackView.arrangedSubviews.isEmpty else {
+            return nil
+        }
+
+        return stackView.arrangedSubviews[0].frame.size.height
+    }
+
+    /// Initializes a view that produces an infinite scrolling effect.
+    /// - Parameter viewBuilder: Closure that generates a `UIView`. Repeatedly called to fill a `UIStackView` that will be scrolled.
+    convenience init(_ viewBuilder: @escaping ((Int) -> UIView)) {
```

Comment:
It seems to me that for this view to work properly, it has to be constructed with this initializer. But right now, it is possible to write this code:
```swift
let view = InfiniteScrollerView()
let view = InfiniteScrollerView(frame: someFrame)
```
And if we use any of the other initializers, the view won't work as expected.

I tried making `init(_ viewBuilder: @escaping ((Int) -> UIView))` the only available init to use by marking the other inits as `private`,  but it didn't work. `init()` was still accessible even though it was marked `private`!

What do you think?

Reply:
👍

Reply:
Very true @salimbraksa, and this is something I'd considered. I've committed f24ceed1de1069c49d20949b3716af8c31170aad, so please let me know your thoughts.